x-pack/plugins/lens/public/visualizations/xy/add_layer.tsx

```diff
@@ -20,16 +20,27 @@ import { EventAnnotationServiceType } from '@kbn/event-annotation-plugin/public'
 import { AddLayerFunction, VisualizationLayerDescription } from '../../types';
 import { LoadAnnotationLibraryFlyout } from './load_annotation_library_flyout';
 import type { ExtraAppendLayerArg } from './visualization';
+import { SeriesType, XYState, visualizationTypes } from './types';
+import { isHorizontalChart, isHorizontalSeries } from './state_helpers';
+import { getDataLayers } from './visualization_helpers';
 import { ExperimentalBadge } from '../../shared_components';
 
 interface AddLayerButtonProps {
+  state: XYState;
   supportedLayers: VisualizationLayerDescription[];
   addLayer: AddLayerFunction<ExtraAppendLayerArg>;
   eventAnnotationService: EventAnnotationServiceType;
   isInlineEditing?: boolean;
 }
 
+export enum AddLayerPanelType {
+  main = 'main',
+  selectAnnotationMethod = 'selectAnnotationMethod',
+  layerVisualizationType = 'layerVisualizationType',
```

Comment:
nit picking here: we can probably rename it so it adhere with the change in title
```suggestion
  selectVisualizationType = 'selectVisualizationType',
```